### DISCUSSION TITLE
CSak az előzőnapi grafikon működik

### DISCUSSION THREAD

**Discussion by ferenchalmai** [ZsBT/hass-w1000-portal#17] [Q&A]
Sziasztok!
Elég kezdő vagyok a HA-ban. Megpróbáltam a minta leírás alapján beüzemelni a W1000-es kiegészítőt, de kevés sikerrel. Az eneria panelen az aznapin egy nagy minusz oszlop szerepel egy villanyóra állással. Az előző napi az jónak néz ki, ott óránként szépen hozza a fogyasztást. A rezsidémon jópofa de 4 hibát hoz (kideríthetetleneket) és csak a villanyóra állás és a hvg kép működik:
Error while executing automation automation.villanyoraallas_valtozik: Error rendering data template: UndefinedError: 'dict object' has no attribute 'to_state'
Villanyóraállás változik: Error executing script. Error for call_service at pos 1: Error rendering data template: UndefinedError: 'dict object' has no attribute 'to_state'
Template variable error: 'dict object' has no attribute 'to_state' when rendering '{{ trigger.to_state.state|float - trigger.from_state.state|float }}'
és a configban:
Invalid config for [template]: [name] is an invalid option for [template]. Check: template->name. (See /config/configuration.yaml, line 45).
Invalid config for [template]: [name] is an invalid option for [template]. Check: template->name. (See /config/configuration.yaml, line 50).
Invalid config for [template]: [name] is an invalid option for [template]. Check: template->name. (See /config/configuration.yaml, line 65).
Van akinek jól működik?

Köszi, Feró

**Comment by ZsBT** [comment]
Szia! Átkonvertálom kérdéssé.

**Comment by ZsBT** [comment]
Szerintem itt eltérhet a szenzor neve és a rezsidémonos template-ben hivatkozott nevek.
Hogyan néz ki a configuration.yaml-ban a w1000-es szekció riportnevei, illetve a rezsidémonos layout yaml konfigja?

**Comment by ferenchalmai** [comment]
w1000-energy-monitor:
  login_user: !secret w1000-email-address
  login_pass: !secret w1000-password
  url: https://energia.eon-hungaria.hu/W1000
  reports: import,export

template:
    - sensor: 
    - name: "tegnapi áramfogyasztás"
      unit_of_measurement: "kWh"
      icon: mdi:home-lightning-bolt-outline
      state: >
        {{ states("input_number.aramfogyasztas_napi") |int }}
    - name: áramfogyasztás napi Ft
      unit_of_measurement: "Ft"
      icon: mdi:cash
      state_class: measurement
# here comes the magic calculus
      state: >
        {% set kwh = states('input_number.aramfogyasztas_napi')|int %}
        {% set napilimit = 2523 / 365 %}
        {% set rezsiforint = 36 %}
        {% set piaciforint = 70.1 %}
        {% if kwh > napilimit %}
          {{ ( napilimit * rezsiforint + (kwh - napilimit) * piaciforint ) | int }}
        {% else %}
          {{ ( kwh * rezsiforint ) | int }}
        {% endif %}
    - name: áramfogyasztás 31napi Ft
      unit_of_measurement: "Ft"
      icon: mdi:cash-multiple
      state_class: measurement
      state: >
        {{ (states('sensor.aramfogyasztas_napi_ft') | int * 31 / 1000) | round(0)  }}000

**Reply by ferenchalmai** [reply]
Ez de furcsán szurta be, valamit nem így kellett volna...
a riport neve az eonnál az egyszerűség kedvéért import

**Reply by ferenchalmai** [reply]
egyaz egyben másoltam a kódot a wiki-ből

**Reply by ZsBT** [reply]
szia! megkésve bár, de elkavarodtak a githubos leveleim.
az automation.yml is be van illesztve? az triggereli az óraállás beolvasását.

**Comment by ferenchalmai** [comment]
![image](https://user-images.githubusercontent.com/116803060/220725774-87a51c1c-6841-4a6d-bc1e-d26a8e5a3651.png)

**Reply by ferenchalmai** [reply]
Pedig milyen jó lenne :)

**Comment by ferenchalmai** [comment]
Szia!

 Igen beillesztettem, de fúrcsán eltöri a sort a feltételben. Javítottam, vissza húztam egy sorba, de akkor is hosszú neki.
És arra panaszkodik a logban hogy nincs "to.state"

De már ez sem rossz ami működik, gratula hozzá és nagyon várom a fejlesztést. Én elég kezdő vagyok még kicsit nehezen igazodom el a rendszerben, de ha megmondod
mit honnan szedjek ki és küldjek el  szívesen segítek a hiba keresésben.

Pl. az sem volt tiszta, hogy az EON-nál mikor a riportot csinálom fontos-e a mezők felvételi sorrendje. Kell-e mindkét mező, vagy elég a villanyóra állás?
Üdv

Halmai Ferenc
